### ISSUE TITLE
Can't find currently working rector + typo3-rector versions

### ISSUE THREAD

**Issue by growsphere** [sabbelasichon/typo3-rector#3029]
# Bug Report

| Subject        | Details                                                         |
| :------------- | :---------------------------------------------------------------|
| Typo3 Rector version |  v0.11.46             |
| Rector version |  v0.13.5 OR v0.13.4 OR lower             |

We are trying to use rector to upgrade a small typo3 project from typo3 v10 to v11. We installed ssch/typo3-rector and rector/rector over composer. Running a rector set (e.g. LevelSetList::UP_TO_PHP_81) works fine, but running a typo3-rector set (e.g. Typo3LevelSetList::UP_TO_TYPO3_11) throws an error:

```
[ERROR] Cannot autowire service "Ssch\TYPO3Rector\FileProcessor\TypoScript\TypoScriptFileProcessor": argument          
         "$editorConfigParser" of method "__construct()" has type "Rector\FileFormatter\EditorConfig\EditorConfigParser"
         but this class was not found.
``` 

The EditorConfigParser Class can indeed not be found in rector v0.13.5 or v.0.13.4. We searched for a version that contains this class and found for example 0.12.23. This version however gives another error when trying to run rector process:
```
[ERROR] Cannot autowire service "Ssch\TYPO3Rector\FileProcessor\TypoScript\TypoScriptFileProcessor": argument          
         "$typoscriptParser" of method "__construct()" references interface                                             
         "Helmich\TypoScriptParser\Parser\ParserInterface" but no such service exists. Did you create a class that      
         implements this interface?
```

We have tried many different versions of typo3 rector and rector, but can't find a currently working set.

We want to run at least:
```
return static function (RectorConfig $rectorConfig): void {
    $rectorConfig->paths([
        __DIR__ . '/src'
    ]);

    // register a single rule
    $rectorConfig->rule(InlineConstructorDefaultToPropertyRector::class);

    // define sets of rules
    $rectorConfig->sets([
        LevelSetList::UP_TO_PHP_81, // works
        \Ssch\TYPO3Rector\Set\Typo3SetList::TYPO3_11 // thows error
    ]);
};
```

We'd appreciate any help!

**Expected behavior:**
First:
```
composer req rector/rector
./vendor/bin/rector init
composer req ssch/typo3-rector
```

Second: change rector php to:
```
return static function (RectorConfig $rectorConfig): void {
    $rectorConfig->paths([
        __DIR__ . '/src'
    ]);

    // register a single rule
    $rectorConfig->rule(InlineConstructorDefaultToPropertyRector::class);

    // define sets of rules
    $rectorConfig->sets([
        LevelSetList::UP_TO_PHP_81,
        \Ssch\TYPO3Rector\Set\Typo3LevelSetList::UP_TO_TYPO3_11,
    ]);
};

```
Finally:
`./vendor/bin/rector process path/to/ext`
--> applies changes without error

**Comment by helsner** [comment]
Hi, pls check the installation docs for full advice there.
In case there is something not clear you can come back to us and report, so we could improve that!

Other than that:
I recommend to install version 0.13.0 of `rector/rector`.
There is no need to install typo3-rector separately, this could even cause issues like you experienced.
From there on you can use the init task as mentioned in the Docs as well.
`vendor/bin/rector init --template-type=typo3 ` will create you a typo3 based rector.php file.

**Comment by samsonasik** [comment]
typo3 no longer included in rector scoped, ref https://github.com/rectorphp/rector-src/pull/2446

it seems this package now need to require rector/rector

**Comment by growsphere** [comment]
With the above suggestion from @helsner reply above I was able to get it to work for now. Thank you!

I want to summarize my findings in case it helps someone else:

* using rector/rector in its current version (0.13.5) and trying `vendor/bin/rector init --template-type=typo3` with it **does not work**. It does generate a rector.php but it is kind of empty and has no Typo3 Set Lists (It also warns that --type is deprecated).
This had led me to believe I needed typo3-rector additionally. For 0.13.5 to work for typo3 upgrade something probably still needs to be implemented or changed.
* using rector/rector with 0.13.0 or 0.13.4 (I chose to use the latter), `vendor/bin/rector init --template-type=typo3` does work and includes the `Typo3LevelSetList::UP_TO_TYPO3_11` set.
* as @helsner suggested, it is NOT necessary (with rector/rector 0.13.4 or lower) to install typo3-rector. typo3-rector is included as a package in rector/rector 0.13.4 or lower, which partly lead to my confusion as it is not evident from the composer.json for instance. I had mistakenly assumed that typo3-rector would require and build upon rector/rector. This dependency separation seems to be a work in progress :)

Thank you all for your feedback.

**Comment by helsner** [comment]
You got everything right there!
With 0.13.4 there seem to be issues with TypoScript processing and similar.
So 0.13.0 is the way to go

0.13.5 does not include typo3-rector which means this package is currently work in progress.
We are not embedded anymore and will build on top of rector as you expected
